en/assembly/mount_gps_compass.md

```diff
@@ -1,15 +1,15 @@
-# Mounting a GPS/Compass
+# Mounting a Compass (or GNSS/Compass)
 
-GPS/Compasses should be mounted on the frame as far away from other electronics as possible, with the direction marker pointing towards the front of the vehicle.
-If mounted in this way you can immediately proceed to [compass calibration](../config/compass.md#performing-the-calibration).
+External compasses, including combined GNSS/Compass hardware, should be mounted on the frame as far away from other electronics as possible, preferrably with the direction marker pointing towards the front of the vehicle (this is the default orientation).
 
-The diagram below shows the heading marker on the Pixhawk 4 and compass.
+The diagram below shows the heading marker on the Pixhawk 4 flight controller and compass.
 
 ![Connect compass/GPS to Pixhawk 4](../../assets/flight_controller/pixhawk4/pixhawk4_compass_gps.jpg)
 
+
 ## Compass Orientation
 
-The compass can be mounted in any of the standard MAVLink orientations defined in [MAV_SENSOR_ORIENTATION](https://mavlink.io/en/messages/common.html#MAV_SENSOR_ORIENTATION).
+If you can't mount it towards the front of the vehicle then it can be mounted in any of the standard MAVLink orientations defined in [MAV_SENSOR_ORIENTATION](https://mavlink.io/en/messages/common.html#MAV_SENSOR_ORIENTATION).
 The orientation follows the same frame convention as when [orienting the flight controller](../config/flight_controller_orientation.md#calculating-orientation).
 
```

Comment:
@sfuhrer It says here that the compass must be mounted in a standard orientation, and below you can see it says "You must mount the compass in a supported orientation!", and also that the orientations are _autodetected_.

BUT if you look at the `CAL_MAG1_ROT` there is option for a custom orientation. Further, if you look at the [advanced FC orientation](https://docs.px4.io/main/en/advanced_config/advanced_flight_controller_orientation_leveling.html) it looks like the FC allows more than these orientations, so it is possible that the mag does too - for example via `CAL_MAG1_PITCH`

So my assertion is that you don't have to use the defined angles - right? Is it still true though that if you don't use those angles the correct orientation would be auto detected?

Reply:
The auto detection as well as the custom angle are fairly new features, and the docs are outdated here.
The flow is now that you only have to set the autopilot rotation, and then do a mag cal. The calibration then detects the mag orientation and sets it. AFAIK you could then override that either by setting the _ROT yourself to the correct value (e.g. "pitch 90"), or set CAL_MAG1_PITCH, CAL_MAG1_ROLL and CAL_MAG1_YAW, which then automatically sets CAL_MAG1_ROT to "Custom Euler Angle". @bresch correct?

Reply:
I'm going to try fix this in https://github.com/PX4/PX4-user_guide/pull/3107 and then rebase. Don't want to block on this.